File: QE-selenium python/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/dynamic-content")
print(driver.title)
driver.find_element(By.XPATH, "//button[text()='Click me!']").click()
wait = WebDriverWait(driver, 10)
text = wait.until(EC.visibility_of_element_located((By.ID, "dynamicText")))
print(text.text)
driver.quit()

File: QE selenium java/Activity12.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity12 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/dynamic-content");
        System.out.println(driver.getTitle());
        driver.findElement(By.xpath("//button[text()='Click me!']")).click();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        WebElement text = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("dynamicText")));
        System.out.println(text.getText());
        driver.quit();
    }
}

